### DISCUSSION TITLE
Will UBlock really ends when Google release Manifest V3?

### DISCUSSION THREAD

**Discussion by ghost** [uBlockOrigin/uBlock-issues#1792] [Q&A]
I just wanted to ask, if UBlock ends when Google releases Manifest V3, or if UBlock releases some update that will fix it.

**Comment by ghost** [comment]
Still not possible to implement some features - https://github.com/uBlockOrigin/uBlock-issues/issues/338#issuecomment-873602429

**Comment by u-RraaLL** [comment]
> I just wanted to ask, if UBlock ends when Google releases Manifest V3, or if UBlock releases some update that will fix it.

uBO isn't a Chrome-only extension. In fact, Chrome isn't even [where it performs best](https://github.com/gorhill/uBlock/wiki/uBlock-Origin-works-best-on-Firefox), so no - uBO won't end **when Google removes Manifest V2** (V3 is already here, has been for months /since February?).

Of course, like @gwarser pointed out with "still" - at the moment a fully functional uBO isn't possible on V3, but that doesn't necessarily mean it won't be by the time V2 gets removed.

**Comment by neroux** [comment]
I admittedly lost somewhat track of this situation. Is there already more clarify on what is going to happen, in particular in the context of the extension? The latest update was Google planned to delay the interface change but they are nonetheless commited to it.

In 2019, @gorhill put it relatively straightforward.

> [I won't tell people what to do. I am pointing out that removing the blocking ability of the webRequest API means the death of uBO, I won't work to make uBO less than what it is now.](https://github.com/uBlockOrigin/uBlock-issues/issues/338#issuecomment-456179825)

That would suggest there won't be uBO any more on Chrome, once Google dropped the old interfaces. Nonetheless there is uBO Lite, which does exactly that. Has the stance changed?

Insight very much appreciated.

**Reply by u-RraaLL** [reply]
The possibilities with V3 grew since then. And gorhill also changed his mind and is already working on the "lesser" uBO version. It's called uBO Lite (uBOL).

Here are the differences: https://www.reddit.com/r/uBlockOrigin/comments/1067als/comment/j3h00xj/

**Reply by neroux** [reply]
Thanks for the response. Could you elaborate on what Google fundamentally changed? I am asking because I am under the impression they made a few tweaks, but there was no fundamental change and Raymond's summary seems to confirm that.

> * Filter lists update only when the extension updates (no fetching up to date lists from servers)
> * Many filters are [dropped at conversion time](https://github.com/gorhill/uBlock/blob/master/dist/mv3/log.txt) due to MV3's limited filter syntax
> * No [crafting your own filters](https://github.com/gorhill/uBlock/wiki/Dashboard:-My-filters) (thus no [element picker](https://github.com/gorhill/uBlock/wiki/Element-picker))
> * No [strict-blocked](https://github.com/gorhill/uBlock/wiki/Strict-blocking) pages
> * No [per-site switches](https://github.com/gorhill/uBlock/wiki/Per-site-switches)
> * No [dynamic filtering](https://github.com/gorhill/uBlock/wiki/Blocking-mode)
> * No [importing external lists](https://github.com/gorhill/uBlock/wiki/Dashboard:-Filter-lists#3rd-party-filter-lists)

It's of course not my call, but I'd regret if he gave in and did decide

> to make uBO less than what it is now

**Reply by uBlock-user** [reply]
It's called uBO Lite, it will always be "less" than uBO is. There's no uBO for mv3 though.

**Reply by neroux** [reply]
@uBlock-user, sure, but the statement was

> I won't work to make uBO less than what it is now

That's the detail I regret.

**Reply by uBlock-user** [reply]
@neroux Technically it is still not. It's called "uBO Lite" which is mv3-only extension.

**Reply by neroux** [reply]
Please don't get me wrong, but I wouldn't want to nitpick. My question was if there was a substantial shift in Google's approach and that does not seem to be the case. And it is that context in which I regret the shift in Raymond's position as far as supporting that is concerned.

**Reply by uBlock-user** [reply]
> if there was a substantial shift in Google's approach and that does not seem to be the case.

Ofcourse it is there. You're not seeing it because almost about 50 % of uBO features are already present. But there's a heavy lid on "innovation" and new possibilities which will arise in uBO in years to come, which will not happen for uBO Lite and not to mention the mv-3 limits on specific APIs which will only work as an obstacle. uBO Lite is a good enough blocker, but it will never be a full-fledged content blocker like uBO is.

**Reply by neroux** [reply]
> Ofcourse it is there

That was precisely my question and that's very unfortunate of course.

> uBO Lite is a good enough blocker, but it will never be a full-fledged content blocker like uBO is

Why I regret that Raymond changed his stance. He was originally adamant not to back the new interface and that apparently changed. I think it would have been a good choice to "boycott" that.

**Reply by u-RraaLL** [reply]
> Why I regret that Raymond changed his stance. He was originally adamant not to back the new interface and that apparently changed. I think it would have been a good choice to "boycott" that.

Originally, Mv3 was even more limiting. And leaving millions of Chrome's uBO users entirely unprotected wouldn't have been a boycott. It'd be precisely what Google wanted.

Yes, the right choice for them would've been the move to Firefox, but it'd be unlikely to happen for the majority of people. Plus, there would still be people who would've love to move, but couldn't. Company or school computers.

**Reply by neroux** [reply]
> And leaving millions of Chrome's uBO users entirely unprotected wouldn't have been a boycott

But this was the original stance and statement.

> Originally, Mv3 was even more limiting

Hence my earlier question of what fundamentally changed. The conclusion seems to be Google made some minor tweaks but the issue itself is unchanged, hence why there is the Lite version, which is contradicting mentioned stance however.

The point is, Google is drastically changing the interfaces and what's possible. Because of that Raymond suggested mentioned "boycott" and this is where he seemingly changed his stance - that's what I find a pity, that's it.

**Reply by bmuddassir** [reply]
"Your browser is managed by your organization." 

This is being shown in Chrome on Windows. 

[ExtensionManifestV2Availability](https://chromeenterprise.google/policies/?policy=ExtensionManifestV2Availability) is visible. 

Is this "managed by your organization" due to UBO?

**Reply by gothic-bum** [reply]
> Is this "managed by your organization" due to UBO?

It will be shown if `ExtensionManifestV2Availability` policy is enabled.

**Reply by bmuddassir** [reply]
![image](https://github.com/user-attachments/assets/ba352da8-c984-46ac-ac4a-95a11c39ba40)

I guess it's enabled with value 2.

**Reply by gothic-bum** [reply]
Per the [documention](https://chromeenterprise.google/policies/?policy=ExtensionManifestV2Availability) for `ExtensionManifestV2Availability` that you previously linked to:
  
> 0 = Default browser behavior
> 1 = Manifest v2 is disabled
> 2 = Manifest v2 is enabled
> 3 = Manifest v2 is enabled for forced extensions only
